device/linux/input/input.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright Antioch. All rights reserved.
+
+#include "hal/input.h"
+
+#include <iostream>
+#include <mutex>
+#include <thread>
+
+namespace hal::input {
+
+static input_callbacks* cb = nullptr;
+std::thread* input_thr;
+
+void input_loop() {
+  std::string s;
+  while (1) {
+    std::cin >> s;
+    std::cout << "Got input \"" << s << "\"" << std::endl;
+    if (s == "ls") {
+      cb->on_left_short_press();
+    }
+    if (s == "ll") {
+      cb->on_left_long_press();
+    }
+    if (s == "rs") {
+      cb->on_right_short_press();
+    }
+    if (s == "rl") {
+      cb->on_right_long_press();
+    }
+    if (s == "ss") {
+      cb->on_select_short_press();
+    }
+    if (s == "sl") {
+      cb->on_select_long_press();
+    }
+  }
+}
+
+input_err_t input_init(input_callbacks* callbacks) {
+  cb = new input_callbacks(*callbacks);
+  input_thr = new std::thread(input_loop);
+  input_thr->detach();
+  return RES_OK;
+}
+
+input_err_t input_destruct() {
+  delete cb;
```

Comment:
Shouldn't we delete the CB only after the input loop stops or are we ok with the potential default here

Reply:
this impl is missing a lot of pieces. this is really just a demo to make sure it sort of works. let me think a bit more about this later if we want to productionize it